ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.cpp

```diff
@@ -0,0 +1,294 @@
+*              See accompanying file LICENSE.txt or
+*              http://www.opengeosys.org/project/license
+*
+*/
+
+#include "TwoPhaseFlowWithPrhoMaterialProperties.h"
+#include <logog/include/logog.hpp>
+#include "BaseLib/reorderVector.h"
+#include "MaterialLib/Fluid/FluidProperty.h"
+#include "MaterialLib/PorousMedium/Porosity/Porosity.h"
+#include "MaterialLib/PorousMedium/Storage/Storage.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CapillaryPressureSaturation.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/CapillaryPressure/CreateCapillaryPressureModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/CreateRelativePermeabilityModel.h"
+#include "MaterialLib/PorousMedium/UnsaturatedProperty/RelativePermeability/RelativePermeability.h"
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/NewtonRaphson.h"
+#include "ProcessLib/Parameter/Parameter.h"
+#include "ProcessLib/Parameter/SpatialPosition.h"
+namespace ProcessLib
+{
+namespace TwoPhaseFlowWithPrho
+{
+TwoPhaseFlowWithPrhoMaterialProperties::TwoPhaseFlowWithPrhoMaterialProperties(
+    bool const has_material_ids,
+    MeshLib::PropertyVector<int> const& material_ids,
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+        liquid_density,
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+        viscosity,
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+        gas_density,
+    std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+        gas_viscosity,
+    std::vector<Eigen::MatrixXd>
+        intrinsic_permeability_models,
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Porosity>>&&
+        porosity_models,
+    std::vector<std::unique_ptr<MaterialLib::PorousMedium::Storage>>&&
+        storage_models,
+    std::vector<std::unique_ptr<
+        MaterialLib::PorousMedium::CapillaryPressureSaturation>>&&
+        capillary_pressure_models,
+    std::vector<
+        std::unique_ptr<MaterialLib::PorousMedium::RelativePermeability>>&&
+        relative_permeability_models)
+    : _has_material_ids(has_material_ids),
+      _liquid_density(std::move(liquid_density)),
+      _viscosity(std::move(viscosity)),
+      _gas_density(std::move(gas_density)),
+      _gas_viscosity(std::move(gas_viscosity)),
+      _material_ids(material_ids),
+      _intrinsic_permeability_models(intrinsic_permeability_models),
+      _porosity_models(std::move(porosity_models)),
+      _storage_models(std::move(storage_models)),
+      _capillary_pressure_models(std::move(capillary_pressure_models)),
+      _relative_permeability_models(std::move(relative_permeability_models))
+{
+    DBUG("Create material properties for Two-Phase flow with PP model.");
+}
+
+void TwoPhaseFlowWithPrhoMaterialProperties::setMaterialID(
+    const ProcessLib::SpatialPosition& pos)
```

Comment:
I think it would be better to pass the material id here. Spatial position might not have the element id set, which is not checked in the function's body.